test/unit/codegen/codegen_cpp_visitor.cpp

```diff
@@ -637,28 +639,39 @@ SCENARIO("Check codegen for MUTEX and PROTECT", "[codegen][mutex_protect]") {
         std::string const nmodl_text = R"(
             NEURON {
                 SUFFIX TEST
-                RANGE tmp
+                RANGE tmp, foo
             }
             PARAMETER {
                 tmp = 10
+                foo = 20
             }
             INITIAL {
                 MUTEXLOCK
                 tmp = 11
                 MUTEXUNLOCK
                 PROTECT tmp = 12
             }
+            PROCEDURE bar() {
+                PROTECT foo = 21
+            }
         )";
 
         THEN("Code with OpenMP critical sections is generated") {
             auto const generated = get_cpp_code(nmodl_text);
-            std::string expected_code = R"(#pragma omp critical TEST {
+            // critical section for the mutex block
+            std::string expected_code_initial = R"(#pragma omp critical (TEST)
+                {
                     inst->tmp[id] = 11.0;
                 }
-                #pragma omp critical TEST {
-                    inst->tmp[id] = 12.0;
-                })";
-            REQUIRE_THAT(generated, Contains(expected_code));
+                #pragma omp atomic update
+                inst->tmp[id] = 12.0;)";
+
+            // atomic update for the PROTECT construct
+            std::string expected_code_proc = R"(#pragma omp atomic update
+        inst->foo[id] = 21.0;)";
```

Comment:
This isn't an atomic update. [The OpenMP spec](https://www.openmp.org/spec-html/5.0/openmpsu95.html) says:
```
...
where expression-stmt is an expression statement with one of the following forms:
...
* If atomic-clause is update or not present:
  x++; 
  x--; 
  ++x; 
  --x; 
  x binop= expr; 
  x = x binop expr; 
  x = expr binop x;
```
I'm not sure if this test code is actually compiled (I think not?), but we should generate something correct.

Reply:
> if this test code is actually compiled

yes, it's not compiled (we just compare the code structure) but yeah it was my sloppiness here!